python_legacy/iceberg/api/expressions/literals.py

```diff
@@ -414,6 +414,8 @@ def to(self, type_var):  # noqa: C901
                     return DecimalLiteral(Decimal(str(self.value))
                                           .quantize(Decimal("." + "".join(["0" for i in range(1, type_var.scale)]) + "1"),
                                                     rounding=ROUND_HALF_UP))
+        elif type_var.type_id == TypeID.BOOLEAN and self.value.upper() in ['TRUE', 'FALSE']:
```

Comment:
What about throwing exception when value is either true or false?

Reply:
thanks @puchengy for checking ,it will by default raise the following, let me know if it works.
`iceberg.exceptions.exceptions.ValidationException: Invalid Value for conversion to type boolean: "false1" (StringLiteral)`

Reply:
got it, approve. Thanks